plugins/modules/backup_selection.py

```diff
@@ -0,0 +1,304 @@
+from ansible_collections.amazon.aws.plugins.module_utils.backup import get_plan_details
+from ansible.module_utils.common.dict_transformations import snake_dict_to_camel_dict
+from ansible.module_utils.common.dict_transformations import camel_dict_to_snake_dict
+
+
+def check_for_update(current_selection, backup_selection_data, iam_role_arn):
+    update_needed = False
+    if current_selection[0].get("IamRoleArn", None) != iam_role_arn:
+        update_needed = True
+
+    fields_to_check = [
+        {
+            "field_name": "Resources",
+            "field_value_from_aws": json.dumps(current_selection[0].get("Resources", None), sort_keys=True),
+            "field_value": json.dumps(backup_selection_data.get("Resources", []), sort_keys=True),
+        },
+        {
+            "field_name": "ListOfTags",
+            "field_value_from_aws": json.dumps(current_selection[0].get("ListOfTags", None), sort_keys=True),
+            "field_value": json.dumps(backup_selection_data.get("ListOfTags", []), sort_keys=True),
+        },
+        {
+            "field_name": "NotResources",
+            "field_value_from_aws": json.dumps(current_selection[0].get("NotResources", None), sort_keys=True),
+            "field_value": json.dumps(backup_selection_data.get("NotResources", []), sort_keys=True),
+        },
+        {
+            "field_name": "Conditions",
+            "field_value_from_aws": json.dumps(current_selection[0].get("Conditions", None), sort_keys=True),
+            "field_value": json.dumps(backup_selection_data.get("Conditions", []), sort_keys=True),
+        },
+    ]
+    for field_to_check in fields_to_check:
+        if field_to_check["field_value_from_aws"] != field_to_check["field_value"]:
+            if (
+                field_to_check["field_name"] != "Conditions"
+                and field_to_check["field_value_from_aws"] != "[]"
+                and field_to_check["field_value"] != "null"
+            ):
+                # advanced settings to be updated
+                update_needed = True
+            if (
+                field_to_check["field_name"] == "Conditions"
+                and field_to_check["field_value_from_aws"]
+                != '{"StringEquals": [], "StringLike": [], "StringNotEquals": [], "StringNotLike": []}'
+                and field_to_check["field_value"] != "null"
+            ):
+                update_needed = True
+
+    return update_needed
+
+
+def main():
+    argument_spec = dict(
+        backup_selection_name=dict(type="str", required=True, aliases=["selection_name"]),
+        backup_plan_name=dict(type="str", required=True, aliases=["plan_name"]),
+        iam_role_arn=dict(type="str"),
+        resources=dict(type="list", elements="str"),
+        conditions=dict(type="dict"),
+        not_resources=dict(type="list", elements="str"),
+        list_of_tags=dict(type="list", elements="dict"),
```

Comment:
I think this should define options rather than being a pure list of freeform dicts...

Reply:
Specified the options in the arg spec.